pkg/storage/db.go

```diff
@@ -473,8 +475,125 @@ func (db *DB) Store(ctx context.Context, cloudAssetChanges domain.CloudAssetChan
 		}
 		return err
 	}
-	return tx.Commit()
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+	ver, err := db.GetSchemaVersion(ctx)
+	if err != nil || ver < DualWriteSchemaVersion {
+		return nil
+	}
+	return db.StoreV2(ctx, cloudAssetChanges)
+}
 
+// StoreV2 an implementation of the Storage interface that records to a database using new schema, to replace Store
+func (db *DB) StoreV2(ctx context.Context, cloudAssetChanges domain.CloudAssetChanges) error {
```

Comment:
per the function comment, do we need to add this method to the Storage interface?

Reply:
It's supposed to replace v1 in-place once we completely move over to new schema.
The primary reason to make it public right now is to be able to test it separately from the existing one.

Reply:
I probably should have reflected this in the comment :)